InciCsvCorrector.java

```diff
@@ -0,0 +1,159 @@
+                boolean currentLineHasValidEnd = isDate(line.substring(line.length() - DATE_LENGTH));              //valid end of line (dd/mm/yyyy)
+                if (!(currentLineHasValidStart && currentLineHasValidEnd)) {               //adds to list the invalid lines
+                    invalidLines.add(lineNum);
+                }
+            }else{
+                invalidLines.add(lineNum);
+            }
+        }
+        return invalidLines;
+    }
+
+    /**
+     * eliminates line endings that break the ingredient entries writing the results in the corrected file
+     * @param source source file
+     * @param corrected result file
+     */
+    private static void correctInterruptedLines(File source,File corrected) throws IOException {
+
+        BufferedReader br = new BufferedReader(new FileReader(source));
+        BufferedWriter bw = new BufferedWriter(new FileWriter(corrected));
+        String line;
+        boolean currentLineHasValidStart;
+        boolean currentLineHasValidEnd;
+        boolean lineCompleted = true;
+
+        while((line = br.readLine())!=null){
+            if(line.length()>MINIMUM_VALID_LENGTH) {
+                currentLineHasValidStart = (isNumeric(line.substring(0, 5)) && (line.charAt(5) == ','));        //valid start of line (12345,)
+                currentLineHasValidEnd = isDate(line.substring(line.length() - DATE_LENGTH));              //valid end of line (dd/mm/yyyy)
+                if (currentLineHasValidStart && currentLineHasValidEnd) {               //line is complete
+                    bw.write(line + "\r\n");          //write and go to the next
+                } else if (currentLineHasValidStart) {         //line has a valid start but is incomplete
+                    bw.write(line);                 //write without ending
+                    lineCompleted = false;
+                } else if (isDate(line.substring(line.length() - DATE_LENGTH)) && !lineCompleted) {   //valid end of line and last line is incomplete
+                    bw.write(line + "\r\n");                                                 //write and end line
+                    lineCompleted = true;
+                } else if (!lineCompleted) {          //last line incomplete
+                    bw.write(line);                 //write without ending line
+                }
+            }
+        }
+        bw.close();
+    }
+
+    /**
+     * gets the file paths from console input
+     * @return String[2] containing the paths of the source file [0] and the new corrected file [1]
+     */
+    private static String[] getFilesPaths(){
+
+        String[] paths = new String[2];
+        Scanner sc = new Scanner(System.in);
+
+        System.out.print("Source file path (C:\\Users\\...) :");
+        paths[0]= sc.nextLine();
+        paths[0]= paths[0].replace("\\","\\\\"); // replace \ with \\ so the path can be used ("\" alone is an escape character)
```

Comment:
You don't have to replace the backslashes in the path, because if you pass the path, without modifying it, of the file to the File constructor, the object will be built without problems .
So this method isn't useful

Reply:
I've tried to do a test modifying the main and removing this class, and the program do all the stuff it have to do

Reply:
Ok, didn't know that.